Project description in README.

alonweissfeld opened the pull request:
Added some description to our project including an overall structure of the platform.

Commit:
Added project desription in README.

lmilbaum commented on README.md:
```diff
@@ -1,3 +1,19 @@
+
+As a podcaster/streamer, you can create your private radio. By accessing your user endpoint, anyone who wants to listen can join your podcast, live djset, or whatever it is you want it to be heard.
+
+As a listener, you can follow your favourite streamers and keep up to date with their streams.
```
s/favourite/favorite/

alonweissfeld replied:
Didn't get what you mean :)

rosomri replied:
@alonweissfeld  There is a typo mistake, should be favorite instead of favourite (line 8)

alonweissfeld replied:
Isn't that a British vs American english thing? anyway fixed.

lmilbaum replied:
This is how you substitute a string by a string in regex. Look it up.

Commit:
Fixed typo.

Commit:
Elaborate on tech stack.

igoihman commented on README.md:
```diff
@@ -1,3 +1,49 @@
+
+As a listener, you can follow your favorite streamers and keep up to date with their streams.
+
+## Idealogy 
```
Ideology

ifireball commented on README.md:
```diff
@@ -1,3 +1,49 @@
+The platform strives to be a simple social network as users can follow each other and go live whenever they wish to do so. Our philosophy is that there should be no alienation between a streamer and a listener, and we engage and strive to push the interaction between them as possible.
+
+## High-level design
+* Login/register page
```
how about some kind of an interesting landing page for unregistered users?

I suggest you try to describe your application in the terms of the kinds of users that use it and their workflows, instead of describing the individual pages. . This is more along the lines of what you wrote in the introduction above, only with a bit more detail. Such a description can be easily turned later into user stories.

ifireball commented on README.md:
```diff
@@ -1,3 +1,49 @@
+    * Live audio streaming: [Django Channels](https://channels.readthedocs.io/en/latest/introduction.html) is under consideration - the subject is under research.
+
+* Frontend:
+    * React
```
I would recommend not to go with React at this point you can do quite a bit with server-side rendering and a little AJAX, using React will probably take time you don't have and might be an overkill for the application you are describing.

ifireball commented on README.md:
```diff
@@ -1,3 +1,49 @@
+    * Python packages conifuged via `requirements.txt`
+
+* CI/CD:
+    * GitHub Actions workflow to run lint, unit and functional tests with pylint and PyTest.
```
did you consider using type checks with [Mypy](http://mypy-lang.org/)?

alonweissfeld replied:
Looks cool!

ifireball commented on README.md:
```diff
@@ -1,3 +1,49 @@
+    * HTML/Sass
+
+* Environment:
+    * Vagrant (specific virtual box not yet determined)
```
Vagrant is only for development, you sohuld state that.

ifireball commented on README.md:
```diff
@@ -1,3 +1,49 @@
+
+* Environment:
+    * Vagrant (specific virtual box not yet determined)
+    * Python packages conifuged via `requirements.txt`
```
Did you consider how you would ensure version consistency? Also, how would oyu manage updates?

ifireball commented on README.md:
```diff
@@ -1,3 +1,49 @@
+## Technological stack 
+* Backend:
+    * **MVC design pattern with Django framework**.
+    * Database: TBD.
```
Django offers sqlite out of the box for development purposes. If you don't have a particular reason to perfer another database, I'd go with PostgreSQL

WRT the tables below - please include some more details abou what they would contain

ifireball commented on README.md:
```diff
@@ -1,3 +1,49 @@
+        * users table
+        * user_followers table
+        * streams table
+    * Live audio streaming: [Django Channels](https://channels.readthedocs.io/en/latest/introduction.html) is under consideration - the subject is under research.
```
Try to find if there are 3rd party libraries or platforms you could use for this. You're not likely to mange to develop an audio-over-IP solution from scratch in 6-10 weeks!

Think if you can have an MVP for yor application without this!

ifireball reviewed (commented):
Added some general comments and recommentations

Commit:
Include users workflows in the high-level design description.

Commit:
Furhter explain the database structure.

Commit:
Updated the technological stack to include Pipenv, Mypy and further descriptions and/or thoughts on the technical side of the project.

On top of that, refined the overall design of the file (replaced main tech stacks bullets with headlines).

ifireball reviewed (approved):
It still needs some work but since today is the deadline, I'm going to approve and you'll need to fix the remaining issues as you move forward.
